libass dynamic library

scarsty opened the pull request:
The static library is not easy to use because too many other libraries have to be linked manually, this port can build a dynamic library.
This library relies on fribidi, harfbuzz, fontconfig, libiconv.

The .def file is from https://github.com/feliwir/libass-msvc.

scarsty committed:
libass dynamic library
it relies on fribidi, harfbuzz, fontconfig, libiconv

Codiferous commented:
@scarsty Wow! You made really quick work of this change 😄 

I've verified on my machine that this does build a working DLL, using the sample test/application in the libass repository. We're still waiting on a full CI run to verify there are no regressions. I'll report back when we get results!

CoffeeFlux commented:
Is there any reason to build with fontconfig on Windows? Just the DirectWrite backend should be fine - the fontconfig one was a constant headache, which is why it's no longer required.

scarsty commented:
> Is there any reason to build with fontconfig on Windows? Just the DirectWrite backend should be fine - the fontconfig one was a constant headache, which is why it's no longer required.
> 
> Also, is there any chance you could update this to generate the pkg-config file? Should be pretty simple, and I can pr it separately if you'd rather not bother.

You are right, fontconfig is of no use. 
I dont't know how to  generate a pkg-config file, is there any tools?

CoffeeFlux commented:
You're essentially setting the variables in the .pc.in file, calling configure_file on it, and then adding it as an install target. If you're not familiar don't worry about it - i'll pr it separately :)

Codiferous commented on ports/libass/CMakeLists.txt:
```diff
@@ -9,25 +9,48 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
+add_compile_definitions(CONFIG_FREETYPE)
+add_compile_definitions(CONFIG_FRIBIDI)
+add_compile_definitions(CONFIG_HARFBUZZ)
+add_compile_definitions(CONFIG_ICONV)
```
Based on the discussion on this PR, I believe that `CONFIG_FONTCONFIG`, `CONFIG_FREETYPE`, and `CONFIG_ICONV` may be unnecessary. Can someone familiar with the library confirm? I'm happy to make the changes and update the PR, but I want to make sure we get it right 👍

scarsty replied:
I have checked and I think you are right. So the relying, include and link messages about them should also be removed.

CoffeeFlux replied:
Fontconfig and iconv are optional, but freetype needs to be linked. See: https://github.com/libass/libass/blob/master/configure.ac#L136

scarsty replied:
The "CONTROL" file should also be corrected.

Codiferous replied:
Okay, I think I finally got the PR in the correct state! This PR has passed CI, so I'd like to merge it. Could either @scarsty or @CoffeeFlux take a quick look?

CoffeeFlux replied:
I haven't actually tested the build output, but this looks fine to me.

Codiferous replied:
@CoffeeFlux Thanks for the quick review. I've used this port to run the test executable [here](https://github.com/libass/libass/blob/master/test/test.c). It appears to do some things that look correct to me (creates an image with subtitles applied), but this is outside of my area of expertise. 

If the port needs more work, we'll refine it in further PRs. Thanks!

Codiferous committed:
Remove references to iconv and fontconfig

Because this port is currently Windows-only (Fribidi doesn't build on
Linux with vcpkg because of an issue with Meson), we don't want to worry
about FontConfig; DirectWrite should be enough.

Codiferous committed:
Freetype is important!

Codiferous committed:
Remove unnecessary fontconfig -- for real

Codiferous committed:
Remove unnecessary comment
